shared/actions/login/index.js

```diff
@@ -458,69 +438,67 @@ function makeKex2IncomingMap (dispatch, getState, onBack: SimpleCB, onProvisione
         }
       }
 
-      const props = {
+      dispatch(navigateAppend([{
+        selected: 'codePage',
         mapStateToProps,
         onBack: onBack,
         setCodePageMode: mode => dispatch(setCodePageMode(mode)),
         qrScanned: code => cb(code.data),
         setCameraBrokenMode: broken => dispatch(setCameraBrokenMode(broken)),
         textEntered: text => cb(text),
         doneRegistering: () => dispatch(doneRegistering()),
-      }
-
-      dispatch(routeAppend({
-        parseRoute: {
-          componentAtTop: {component: CodePage, props},
-        },
-      }))
+      }]))
```

Comment:
No `parentPath` here because this gets appended to the `devicesTab` during new device provisioning.